internals/init_ecdh_test.go

```diff
@@ -0,0 +1,115 @@
+package internals
+
+import (
+	"globe-and-citizen/layer8/middleware/js"
+	"globe-and-citizen/layer8/middleware/storage"
+	"strings"
+	"testing"
+
+	utilities "github.com/globe-and-citizen/layer8-utils"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitializeECDH(t *testing.T) {
+	serverPri, serverPub, err := utilities.GenerateKeyPair(utilities.ECDH)
+	assert.Nil(t, err)
+	assert.NotNil(t, serverPri)
+	assert.NotNil(t, serverPub)
+
+	storage.InitInMemStorage(serverPri, serverPub)
+	db := storage.GetInMemStorage()
+	assert.NotNil(t, db)
+
+	b64ServerPub, err := serverPub.ExportAsBase64()
+	assert.Nil(t, err)
+	assert.NotEmpty(t, b64ServerPub)
+
+	clientPri, clientPub, err := utilities.GenerateKeyPair(utilities.ECDH)
+	assert.Nil(t, err)
+	assert.NotNil(t, clientPri)
+	assert.NotNil(t, clientPub)
+
+	b64ClientPub, err := clientPub.ExportAsBase64()
+	assert.Nil(t, err)
+	assert.NotEmpty(t, b64ClientPub)
+
+	shared, err := serverPri.GetECDHSharedSecret(clientPub)
+	assert.Nil(t, err)
+	assert.NotNil(t, shared)
+
+	b64Shared, err := shared.ExportAsBase64()
+	assert.Nil(t, err)
+	assert.NotEmpty(t, b64Shared)
+
+	mpjwt, err := utilities.GenerateStandardToken(uuid.New().String())
+	assert.Nil(t, err)
+	assert.NotEmpty(t, mpjwt)
+
+	type args struct {
+		headers *js.Value
+	}
+	tests := []struct {
+		name          string
+		args          args
+		wantShared    string
+		wantPub       string
+		wantMpJWT     string
+		wantErr       bool
+		wantErrString string
+	}{
+		{
+			name: "init_with_valid_headers",
+			args: args{
+				headers: js.ValueOf(map[string]interface{}{
```

Comment:
To do the testing systematically, my understanding is that the argument to InitializeECDH() becomes the source of variance. It is a map[string]interface{} which implies the following test cases:
```
EMPTY: 
headers: js.ValueOf(map[string]interface{}{}

RANDOM ADDED HEADERS / TOO MANY HEADERS:
headers: js.ValueOf(map[string]interface{}{
    "addedheader": "artibtrarystring"
    "x-ecdh-init": b64ClientPub,
    "x-client-uuid": uuid.New().String(),
     "mp-jwt": mpjwt,
}

DUPLICATE HEADERS:
headers: js.ValueOf(map[string]interface{}{
    "x-ecdh-init": b64ClientPub,
    "x-ecdh-init": b64ClientPub,
    "x-client-uuid": uuid.New().String(),
     "mp-jwt": mpjwt,
}


AND FOR EACH OF THE STRINGS, STRING ABUSES:
headers: js.ValueOf(map[string]interface{}{
     "x-ecdh-init":   b64ClientPub | invalid characters | <verylongstring> | <multiline\nstring> | ""
     "x-client-uuid": uuid.New().String() | invalid characters | <verylongstring> | <multiline\nstring> | ""
     "mp-jwt":        mpjwt | invalid characters | <verylongstring> |  <multiline\nstring> | ""
}
```

Reply:
![image](https://github.com/globe-and-citizen/layer8-middleware/assets/116566901/ad613acb-2963-4b36-9db2-b34871b46fe8)

Reply:
Good point! I'll add these cases